maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/FileFinder.java

```diff
@@ -177,14 +150,22 @@ private Set<String> exclude(List<FileMatcher> excludes, Set<FileRequest> locally
     }
 
     private boolean isValid(FileRequest file,
-                            List<FileMatcher> excludes) {
+                                     List<FileMatcher> excludes,
+                                     List<FileMatcher> includes) {
 
         // file may have been deleted from GIT (or locally)
         if (!Files.exists(file.path())) {
             Log.debug("File " + file.relativePath() + " does not exist, ignoring.");
             return false;
         }
 
+        for (FileMatcher include : includes) {
```

Comment:
> An optional prefix "!" which negates the pattern; any matching file excluded by a previous pattern will become included again. It is not possible to re-include a file if a parent directory of that file is excluded. Git doesn’t list excluded directories for performance reasons, so any patterns on contained files have no effect, no matter where they are defined. Put a backslash ("\") in front of the first "!" for patterns that begin with a literal "!", for example, "\!important!.txt".

From the git documentation https://git-scm.com/docs/gitignore. So I used an `includes` list to re include the file if necessary. This logic belongs in the `Gitignore` class instead of `FileFinder`, I will refactor it.

Reply:
roger that.